### PR TITLE
Ecto rewrite

### PR DESCRIPTION
**Pull request by rsierra**
This is just a rewrite of the macros implementation following the ecto style.

### CODE REVIEW

File: lib/decoratex.ex [hunk 1]
```diff
@@ -120,19 +120,42 @@ defmodule Decoratex do
...
+      Module.register_attribute(__MODULE__, :decorations, accumulate: true)
+
+      try do
+        import Decoratex
+        unquote(block)
+      after
+        :ok
+      end
+
+      decorations = @decorations |> Enum.reverse()
```
Comment (odarriba) [inline lib/decoratex.ex]: I wouldn't use pipes if there is only one call

File: lib/decoratex.ex [hunk 2]
```diff
@@ -149,7 +172,7 @@ defmodule Decoratex do
 
       @spec decorate(struct) :: struct
       def decorate(element) do
-        element.__struct__.__decorations__()
+        @decorations
```
Comment (odarriba) [inline lib/decoratex.ex]: Same here

File: lib/decoratex.ex [hunk 3]
```diff
@@ -164,8 +187,12 @@ defmodule Decoratex do
       def decorate(element, name) when is_atom(name), do: decorate(element, [name])
 
       @spec decorate(struct, except: list(atom)) :: struct
-      def decorate(element, except: names) when is_list(names) do
-        decorate(element, Map.keys(element.__struct__.__decorations__()) -- names)
+      def decorate(element, except: exceptions) when is_list(exceptions) do
+        names =
+          @decorations
```
Comment (odarriba) [inline lib/decoratex.ex]: And here

**Review by odarriba — APPROVED** [review]
LGTM

File: lib/decoratex.ex (hunk 1 above)
Comment (rsierra) [reply lib/decoratex.ex]: Umm, i think that in some cases is more readable. Not this one but in the others (this line is from ecto 😅)

**Pull request merged**